src/vs/vscode.d.ts

```diff
@@ -1668,6 +1668,24 @@ declare module 'vscode' {
 	}
 
 	/**
+	 * The type definition provider defines the contract between extensions and
+	 * the go to type definition feature.
```

Comment:
Like the other language features this needs a section the doc (editing evolved) such that people has a clear understanding of what this is